spec/simple_segment/configuration_spec.rb

```diff
@@ -23,4 +23,17 @@
       expect(config.on_error).to be_a(Proc)
     end
   end
+
+  it 'works with stub' do
+    config = described_class.new(write_key: 'test', stub: true)
+    expect(config.stub).to eq true
+  end
+
+  it 'works with user prefered logging' do
+    config = described_class.new(
+      write_key: 'test',
+      logger: Logger.new(STDOUT)
+    )
+    expect(config.logger).not_to be_nil
```

Comment:
I think this does not actually test setting a logger (since we have a default one), can we try doing something like the following?

```ruby
my_logger = object_double('Logger')
config = described_class.new(
  write_key: 'test',
  logger: my_logger
)
expect(config.logger).to eq(my_logger)
```